### PR TITLE
int: Initial monorepo

### PR DESCRIPTION
**Pull request by jamesclark-Zapata**
# The problem

In preparation for separate packages, we require a monorepo setup.

# This PR's solution

Implements the ADR, excluding a [rye workspace](https://rye-up.com/guide/workspaces/) which can be added at any time later.

- `orquestra-sdk` is moved to the `projects/` directory
- Github actions workflows are modified:
  - Generic "coverage" and "style" workflows that are reusable.
  - A single PR workflow that checks for modified files and runs tests/checks.
    This workflow has a single job that can be used as the required check. This only passes if all the style checks/tests that were actually run pass.
  - Updated the nightly check to use the reusable workflows
  - Updates the PR comment action to use the working directory

Note: the nightly workflow hasn't been tested yet because I renamed it and it won't show up in the UI until it's on `main`.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: .github/workflows/all-pr-checks.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      packages: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          # We can codify the path dependencies between the packages here
```
Comment (alexjuda) [inline .github/workflows/all-pr-checks.yml:20]: **suggestion**: it seems this steps produces outputs we depend on later on. Could you add a comment what outputs we expect, and maybe provide example values? 

The [action's documentation](https://github.com/dorny/paths-filter?tab=readme-ov-file#outputs) mentions "filters" and "changes", but we also use `changes` as a GH job name, and `filter` as a step ID, and this naming overlap made it difficult for me to understand our set up.

**Review by alexjuda — APPROVED** [review]
Looks nice! Added non-blocking suggestions

**Review by alexjuda — APPROVED** [review]
Looks nice! Added non-blocking suggestions

File: .github/workflows/all-pr-checks.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    strategy:
+      matrix:
+        package: ${{ fromJSON(needs.changes.outputs.packages) }}
+    uses: ./.github/workflows/package-coverage.yml
+    with:
+      target_path: ${{ matrix.package }}
+      # Choice of Python versions to run against:
+      # * 3.9: the oldest supported version
+      # * 3.11: the latest officially supported version. Used in CE images.
+      python_versions: '["3.9", "3.11"]'
```
[the remark this answers is not in the document]
Comment (jamesclark-Zapata) [reply .github/workflows/all-pr-checks.yml:37]: Yeah. I thought I did this already, but I changed style insted of test 🤦

File: .github/workflows/all-pr-checks.yml (hunk 1 above)
Comment (jamesclark-Zapata) [reply .github/workflows/all-pr-checks.yml:31]: does this help? 4467194

Comment (alexjuda) [reply .github/workflows/all-pr-checks.yml:31]: Ohhh, I can see it now. Yes, it's much easier now, thanks!

**Pull request merged**